### ISSUE TITLE
Automatic array pattern destructing

### ISSUE THREAD

**Issue by diamond95** [sidorares/node-mysql2#1196]
I have connection on database: 

```
const mysql = require('mysql2')
const config = require('../config/config')

const pool = mysql.createPool({
    database: config.db.database,
    user: config.db.user,
    password: config.db.password,
    host: config.db.options.host,
    port: config.db.options.port
    
})

module.exports = pool.promise()

```
So when I'm using call on database, like this:
`const response = await db.query('call get_workflow_configuration(?)', [user_id])`
All data is stored in response, but too many arrays from real data.
Response:
```
[
  [
    [ [TextRow], [TextRow] ],
    ResultSetHeader {
      fieldCount: 0,
      affectedRows: 0,
      insertId: 0,
      info: '',
      serverStatus: 34,
      warningStatus: 0
    }
  ],
  [ [ [ColumnDefinition] ], undefined ]
]
```
So, using destructing pattern, I can access data instead call above like this:
`const [[response]] = await db.query('call get_workflow_configuration(?)', [user_id])`
It's kinda annoying use that in whole project, I'm trying to find a way to automaticly destruct response, it should be like this:
```
const response = await...
console.log(response[0])
``` and here data is.

Any advice?
Thanks

**Comment by sidorares** [comment]
Stored procedure can return multiple results, and each result can be a "table", e.i array of rows. In promise api result of `.query()` is [rows, columns] for single statement result and [[rows1, rows2, ...], [columns1, columns2, ...]] for consistency with callback style `callback(error, rows, columns) {}` signature

You can use array destructuring like this

```js
const [[configurations]] = await db.query('call get_workflow_configuration(?)', [user_id]); // to get first statement rows into configurations array

// or like this to assign first row of first statement
const [[[configuration]]] = await db.query('call get_workflow_configuration(?)', [user_id])
```

`[[[]]]` can be a bit hard to follow, I personally would rather split it into multiple assignments instead of deep destructuring

**Comment by sidorares** [comment]
some examples of possible combinations you can get when calling a stored procedure: https://github.com/sidorares/node-mysql2/blob/96fe54c73dd5d99d308ca7c2d2e1dcf3ddb60d24/test/integration/connection/test-multiple-results.js#L70-L94

**Comment by diamond95** [comment]
> Stored procedure can return multiple results, and each result can be a "table", e.i array of rows or "OK header" - insert/update/delete result. In promise api result of `.query()` is [rows, columns] for single statement result and [[rows1, rows2, ...], [columns1, columns2, ...]] for consistency with callback style `callback(error, rows, columns) {}` signature
> 
> You can use array destructuring like this
> 
> ```js
> const [[configurations]] = await db.query('call get_workflow_configuration(?)', [user_id]); // to get first statement rows into configurations array
> 
> // or like this to assign first row of first statement
> const [[[configuration]]] = await db.query('call get_workflow_configuration(?)', [user_id])
> ```
> 
> `[[[]]]` can be a bit hard to follow, I personally would rather split it into multiple assignments instead of deep destructuring

Yeah, I know that.. But is there any hack how can I automatically remove / do not use destructing?

**Comment by sidorares** [comment]
unfortunately difficult to change now in backwards-compatible way. There are few options: add some extra config flag say "omitFields" or add some new api `connection.queryEx` ( whatever naming is better )

mariadb driver returns and array with additional "meta" property that stores fields - maybe that has a bit better ergonomics.

**Comment by diamond95** [comment]
Allright, thanks
